proxy/src/source.rs

```diff
@@ -31,21 +32,23 @@ pub struct ProjectId {
     pub domain: String,
 }
 
-impl From<radicle_registry_client::ProjectId> for ProjectId {
-    fn from(id: radicle_registry_client::ProjectId) -> Self {
-        Self {
+impl TryFrom<radicle_registry_client::ProjectId> for ProjectId {
+    type Error = &'static str;
+    fn try_from(id: radicle_registry_client::ProjectId) -> Result<Self, Self::Error> {
+        Ok(Self {
             name: id.0.to_string(),
             domain: id.1.to_string(),
-        }
+        })
     }
 }
 
-impl Into<radicle_registry_client::ProjectId> for ProjectId {
-    fn into(self) -> radicle_registry_client::ProjectId {
-        (
+impl TryInto<radicle_registry_client::ProjectId> for ProjectId {
+    type Error = &'static str;
```

Comment:
So I think we want `type Error = String` to keep consistent with [`from_string`](https://github.com/radicle-dev/radicle-registry/blob/master/runtime/src/string32.rs#L17)

Then we can do:
```rust
let project_name = ProjectName::from_string(self.name)?;
let project_domain = ProjectDomain::from_string(self.domain)?;
Ok(project_name, project_domain)
```

Reply:
Perfect, I was just about to ask this!